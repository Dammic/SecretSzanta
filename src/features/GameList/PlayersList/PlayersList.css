@import 'variables/variables.css';

.playersList {
    height: unset;
    flex-grow: 1;
    overflow-y: auto;
    overflow-x: hidden;
    background-color: var(--players-list-bg);
    margin-top: 5px;
    padding: 0 10px 10px 10px;
    color: var(--players-list-font);
}

.category {
    text-align: right;
    height: 20px;
    border-bottom: 1px solid var(--players-list-element-border);
    margin-left: -10px;
    margin-right: -10px;
    padding-top: 5px;
    background-color: var(--players-list-category-bg);

    & > span {
        margin-right: 5px;
        font-size: 13px;
    }
}

.playerRow {
    position: relative;
    display: flex;
    align-items: center;
    padding: 3px;
    border-bottom: 1px solid var(--players-list-element-border);
}

.avatar {
    height: 30px;
    width: 30px;
    border-radius: 50px;
    display: none;
    @media (--tablet) { 
        display: initial; 
    }
}

.playerName {
    margin: 0 10px;
    max-width: 70%;

    @media (--tablet) { 
        display: initial;
        max-width: 50%;
    }
}

.roomName {
    font-size: 12px;
    max-width: 33%;
    position: absolute;
    right: 0px;
    display: none;
    @media (--tablet) { 
        display: initial;
    }

    &.mobile {
        display: unset;
        @media (--tablet) { 
            display: none;
        }
    }
}

.roomNameText {
    font-size: 13px; 
}
