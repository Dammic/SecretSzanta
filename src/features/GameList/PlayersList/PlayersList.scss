.players-list {
    height: 500px;
    background-color: $players-list-bg;
    margin-top: 5px;
    padding: 0 10px 10px 10px;
    color: $players-list-font;

    .category {
        text-align: right;
        height: 20px;
        border-bottom: 1px solid $players-list-element-border;
        margin-left: -10px;
        margin-right: -10px;
        padding-top: 5px;
        background-color: $players-list-category-bg;

        > span {
            margin-right: 5px;
            font-size: 13px;
        }
    }
    .player-row {
        display: flex;
        align-items: center;
        padding: 3px;
        border-bottom: 1px solid $players-list-element-border;
        .avatar {
            height: 30px;
            width: 30px;
            border-radius: 50px;
            display: none;
            @include respond-to(medium) { 
                display: initial; 
            }
        }
        > span {
            margin-left: 10px; 
        }
        .room-name {
            position: absolute;
            right: 30px;
            .small-font {
                font-size: 12px; 
            }
        }
    }
}
