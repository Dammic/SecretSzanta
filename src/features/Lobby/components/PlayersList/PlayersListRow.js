import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { getAvatar } from '../../../../utils/avatarsHelper'

import styles from './PlayersList.module.css'
import commonStyles from '../../../Shared/CommonStyles/commonStyles.module.css'

export const PlayersListRow = ({ playerName, avatarNumber, currentRoom }) => {
    return (
        <div className={classNames(styles.playerRow, { [styles.busy]: !!currentRoom })}>
            <img
                className={styles.avatar}
                src={getAvatar(`liberal-${avatarNumber}`)}
                alt="Player avatar"
            />
            <span className={classNames(commonStyles.ellipsis, styles.playerName)}>{playerName}</span>
            {currentRoom && (
                <Fragment>
                    <span className={classNames(commonStyles.ellipsis, styles.roomName)}>
                        <span>
                            <span>room:{' '}</span>
                            <b className={styles.roomNameText}>{currentRoom}</b>
                        </span>
                    </span>
                    <span className={classNames(styles.roomName, styles.mobile)}>Busy</span>
                </Fragment>
            )}
        </div>
    )
}

PlayersListRow.propTypes = {
    playerName: PropTypes.string.isRequired,
    avatarNumber: PropTypes.number.isRequired,
    currentRoom: PropTypes.string,
}
