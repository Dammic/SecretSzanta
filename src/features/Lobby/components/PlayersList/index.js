export { default as PlayersList } from './PlayersList'
