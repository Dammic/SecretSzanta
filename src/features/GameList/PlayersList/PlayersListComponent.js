import React from 'react'
import PropTypes from 'prop-types'
import { map, filter, reject, size } from 'lodash'
import PlayersListRow from './PlayersListRow'

import styles from './PlayersList.css'

const PlayersListComponent = ({ players }) => {
    return (
        <div className={styles.playersList}>
            <div className={styles.category}>
                <span>Online: <b>{size(players)}</b></span>
            </div>
            {map(filter(players, player => !!player.currentRoom), player => (
                <PlayersListRow
                    key={player.playerName}
                    playerName={player.playerName}
                    avatarNumber={player.avatarNumber}
                    currentRoom={player.currentRoom}
                />
            ))}
            {map(reject(players, player => !!player.currentRoom), player => (
                <PlayersListRow
                    key={player.playerName}
                    playerName={player.playerName}
                    avatarNumber={player.avatarNumber}
                    currentRoom={player.currentRoom}
                />
            ))}
        </div>
    )
}

PlayersListComponent.propTypes = {
    players: PropTypes.objectOf(PropTypes.any),
}

export default PlayersListComponent
