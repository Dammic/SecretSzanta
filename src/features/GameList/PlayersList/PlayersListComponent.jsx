import React from 'react'
import PropTypes from 'prop-types'
import { map, filter, reject } from 'lodash'
import PlayersListRow from './PlayersListRow'

const PlayersListComponent = ({ players }) => {
    return (
        <div className="players-list">
            <div className="category">
                <span>Currently logged players: <b>0</b></span>
            </div>
            {map(filter(players, player => !!player.currentRoom), player => (
                <PlayersListRow
                    key={player.playerName}
                    playerName={player.playerName}
                    avatarNumber={player.avatarNumber}
                    currentRoom={player.currentRoom}
                />
            ))}
            {map(reject(players, player => !!player.currentRoom), player => (
                <PlayersListRow
                    key={player.playerName}
                    playerName={player.playerName}
                    avatarNumber={player.avatarNumber}
                    currentRoom={player.currentRoom}
                />
            ))}
        </div>
    )
}

PlayersListComponent.propTypes = {
    players: PropTypes.arrayOf(PropTypes.object),
}

export default PlayersListComponent
