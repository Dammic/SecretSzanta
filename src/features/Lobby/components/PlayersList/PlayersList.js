import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { map, size, groupBy } from 'lodash'
import { PlayersListRow } from './PlayersListRow'

import styles from './PlayersList.module.css'

export const PlayersList = ({ playersList }) => {
    const groupedPlayers = groupBy(playersList, player => !!player.currentRoom)

    return (
        <div className={styles.playersList}>
            <div className={styles.category}>
                <span>Online: <b>{size(playersList)}</b></span>
            </div>
            {map(groupedPlayers.true, player => (
                <PlayersListRow
                    key={player.playerName}
                    playerName={player.playerName}
                    avatarNumber={player.avatarNumber}
                    currentRoom={player.currentRoom}
                />
            ))}
            {map(groupedPlayers.false, player => (
                <PlayersListRow
                    key={player.playerName}
                    playerName={player.playerName}
                    avatarNumber={player.avatarNumber}
                    currentRoom={player.currentRoom}
                />
            ))}
        </div>
    )
}

PlayersList.propTypes = {
    playersList: PropTypes.objectOf(PropTypes.shape({
        playerName: PropTypes.string.isRequired,
        avatarNumber: PropTypes.number.isRequired,
        currentRoom: PropTypes.string,
    })),
}

const mapStateToProps = ({ lobby }) => ({
    playersList: lobby.playersList,
})

export default connect(mapStateToProps)(PlayersList)
