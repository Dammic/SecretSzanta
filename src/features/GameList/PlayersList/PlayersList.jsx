import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import PlayersListComponent from './PlayersListComponent'

class PlayersList extends PureComponent {
    static propTypes = {
  
   
    }

    render() {
        const fakePlayers = [
            {
                playerName: 'testplayer1',
                currentRoom: 'testroom1',
                avatarNumber: 1,
            },
            {
                playerName: 'testplayer3',
                avatarNumber: 3,
            },
            {
                playerName: 'testplayer2',
                currentRoom: 'testroom2',
                avatarNumber: 2,
            },
        ]
        return (
            <PlayersListComponent
                players={fakePlayers}
            />
        )
    }
}

const mapStateToProps = () => ({

})

export default connect(mapStateToProps)(PlayersList)
