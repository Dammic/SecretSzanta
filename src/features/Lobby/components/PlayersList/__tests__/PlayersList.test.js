import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import { aPlayersList, aPlayer } from 'packages/factories'
import { PlayersList } from '../PlayersList'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const mockPlayersList = aPlayersList(null, { seed: 2 })
    const mockPlayerWithoutRoom = aPlayer({ currentRoom: null }, { seed: 1 })
    mockPlayersList[mockPlayerWithoutRoom.playerName] = mockPlayerWithoutRoom
    console.log(mockPlayerWithoutRoom)
    console.log(mockPlayersList)
    const props = {
        playersList: mockPlayersList,
        ...propsOverrides,
    }
    const component = renderMethod(<PlayersList {...props} />)
    return { props, component }
}

describe('<PlayersList />', () => {
    it('matches snapshot - renders list of 4 players with the ones having a  room at the top', () => {
        const { props } = setupProps()
        expectShallowMatchingSnapshot(<PlayersList {...props} />)
    })
})
