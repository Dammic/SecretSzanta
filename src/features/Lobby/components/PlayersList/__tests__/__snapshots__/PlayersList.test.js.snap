// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayersList /> matches snapshot - renders list of 4 players with the ones having a  room at the top 1`] = `
<div
  className="playersList"
>
  <div
    className="category"
  >
    <span>
      Online: 
      <b>
        4
      </b>
    </span>
  </div>
  <PlayersListRow
    avatarNumber={4}
    currentRoom="Division"
    playerName="Hubert93"
  />
  <PlayersListRow
    avatarNumber={1}
    currentRoom="magenta"
    playerName="Jazmin33"
  />
  <PlayersListRow
    avatarNumber={2}
    currentRoom="payment"
    playerName="Lila48"
  />
  <PlayersListRow
    avatarNumber={1}
    currentRoom={null}
    playerName="Hayden_Zieme0"
  />
</div>
`;
